K8s: 7.2.4-2 release notes and supported distributions (#2763)
RedisLabs/redislabs-docs: content/kubernetes/release-notes/7-2-4-releases/7-2-4-2.md

    /kubernetes/release-notes/7-2-4-releases/7-2-4-2/,
      ]
---

## Highlights

This version offers:

- Support for [Redis Enterprise 7.2.4]({{<relref "/rs/release-notes/rs-7-2-4-releases/rs-7-2-4-52">}})

- New default engine, Speedb, for Auto Tiering (formerly Redis on Flash)

## New in this release

### Enhancements

- Support for [Redis Enterprise 7.2.4]({{<relref "/rs/release-notes/rs-7-2-4-releases/rs-7-2-4-52">}}) (RED-73681)
- New default engine, Speedb, for Auto Tiering (formerly Redis on Flash) (RED-90850)
- Security context hardening (RED-98447)

### Resolved issues

- Fixed: Services or routes from the Active-Active setup might not be removed (RED-77752)
- Fixed: Admission controller may not alert when patching the REAADB with missing secret or RERC (RED-104463)

### API changes

The following fields were added to the RedisEnterpriseCluster (REC) custom resource definition (CRD):

- Added `status.BundledDatabaseRedisVersions.major` as a Boolean field indicating whether the Redis version is major.
- Added `spec.RedisOnFlashSpec.bigStoreDriver` to indicate the storage engine used for Auto Tiering (formerly Redis on Flash).

## Version changes

### Breaking changes

> The following changes included in this release affect the upgrade process. Please read carefully before upgrading to 7.24-2.

Reviewer: ```suggestion
The following changes included in this release affect the upgrade process. Please read carefully before upgrading to 7.2.4-8.
```

> The following changes included in this release affect the upgrade process. Please read carefully before upgrading to 7.24-2.

Author: ```suggestion
The following changes included in this release affect the upgrade process. Please read carefully before upgrading to 7.2.4-2.
```

#### ValidatingWebhookConfiguration

Versions 6.4.2-4 and later include a new `ValidatingWebhookConfiguration` resource to replace the `redb-admission` webhook resource. To use releases 6.4.2-4 or later, delete the old webhook resource and apply the new file. See [upgrade Redis cluster]({{<relref "/kubernetes/re-clusters/upgrade-redis-cluster#reapply-webhook">}}) for instructions.

#### OpenShift SCC

> Versions 6.4.2-6 and later include a new SCC (`redis-enterprise-scc-v2`) that you need to bind to your service account before upgrading. OpenShift clusters running version 6.2.12 or earlier upgrading to version 6.2.18 or later might get stuck if you skip this step. See [upgrade a Redis Enterprise cluster (REC)]({{<relref "/kubernetes/re-clusters/upgrade-redis-cluster#before-upgrading">}}) for expand.

Reviewer: ```suggestion
Versions 6.4.2-6 and later include a new SCC (`redis-enterprise-scc-v2`) that you need to bind to your service account before upgrading. OpenShift clusters running version 6.2.12 or earlier upgrading to version 6.2.18 or later might get stuck if you skip this step. See [upgrade a Redis Enterprise cluster (REC)]({{<relref "/kubernetes/re-clusters/upgrade-redis-cluster#before-upgrading">}}) for instructions.
```

### Supported distributions

The following table shows supported distributions at the time of this release. You can also find this list in [Supported Kubernetes distributions]({{< relref "/kubernetes/reference/supported_k8s_distributions" >}}).

<span title="Check mark icon">&#x2705;</span> Supported – This distribution is supported for this version of Redis Enterprise Software for Kubernetes.

<span title="Warning icon">&#x26A0;&#xFE0F;</span> Deprecated – This distribution is still supported for this version of Redis Enterprise Software for Kubernetes, but support will be removed in a future release.

<span title="X icon">&#x274c;</span> End of life – Support for this distribution ended.

Any distribution not listed below is not supported for production workloads.

| **Kubernetes version** | **1.22** | **1.23** | **1.24** | **1.25** | **1.26** | **1.27** |
|---|---|---|---|---|---|---|
| Community Kubernetes | <span title="X icon">&#x274c;</span> | <span title="X icon">&#x274c;</span> | <span title="Check mark icon">&#x2705;</span> | <span title="Check mark icon">&#x2705;</span> | <span title="Check mark icon">&#x2705;</span> | <span title="Check mark icon">&#x2705;</span> |

> | Amazon EKS | <span title="X icon">&#x274c;</span> | <span title="Warning icon">&#x26A0;&#xFE0F;</span> | <span title="Check mark icon">&#x2705;</span> | <span title="Check mark icon">&#x2705;</span> | <span title="Check mark icon">&#x2705;</span> | <span title="Check mark icon">&#x2705;</span> |

Reviewer: Hover text in all rows
```suggestion
| Amazon EKS | <span title="End of life">&#x274c;</span> | <span title="Deprecated">&#x26A0;&#xFE0F;</span> | <span title="Supported">&#x2705;</span> | <span title="Supported">&#x2705;</span> | <span title="Supported">&#x2705;</span> | <span title="Supported">&#x2705;</span> |
```

| VMware TKG 1.6 | <span title="Warning icon">&#x26A0;&#xFE0F;</span> | <span title="Warning icon">&#x26A0;&#xFE0F;</span> |  |  |  |  |
| VMware TKG 2.1 | <span title="Warning icon">&#x26A0;&#xFE0F;</span> | <span title="Warning icon">&#x26A0;&#xFE0F;</span> | <span title="Check mark icon">&#x2705;</span> |  |  |  |
| VMware TKG 2.2 |  | <span title="Check mark icon">&#x2705;</span> | <span title="Check mark icon">&#x2705;</span> | <span title="Check mark icon">&#x2705;</span> |  |  |
| **OpenShift version** | **4.9** | **4.10** | **4.11** | **4.12** | **4.13** |  |
|  | <span title="X icon">&#x274c;</span> | <span title="Warning icon">&#x26A0;&#xFE0F;</span> | <span title="Check mark icon">&#x2705;</span> | <span title="Check mark icon">&#x2705;</span> | <span title="Check mark icon">&#x2705;</span> |  |
| **VMware TKGI version** | **1.13** | **1.14** | **1.15** | **1.16** |  |  |
|  | <span title="X icon">&#x274c;</span> | <span title="Warning icon">&#x26A0;&#xFE0F;</span> | <span title="Check mark icon">&#x2705;</span> | <span title="Check mark icon">&#x2705;</span> |  |  |

## Downloads

- **Redis Enterprise**: `redislabs/redis:7.2.4-52`

> - **Operator**: `redislabs/operator:7.2.4-2`
> - **Services Rigger**: `redislabs/k8s-controller:7.2.4-2.`

Reviewer: Build number?
```suggestion
- **Operator**: `redislabs/operator:7.2.4-8`
- **Services Rigger**: `redislabs/k8s-controller:7.2.4-8`
```

#### OpenShift images

- **Redis Enterprise**: `registry.connect.redhat.com/redislabs/redis-enterprise:7.2.4-52.rhel8-openshift`
    (or `redislabs/redis-enterprise:7.2.4-52.rhel7-openshift` if upgrading from RHEL 7)

> - **Operator**: `registry.connect.redhat.com/redislabs/redis-enterprise-operator:7.2.4-2`
> - **Services Rigger**: `registry.connect.redhat.com/redislabs/services-manager:7.2.4-2`

Reviewer: Build number?
```suggestion
- **Operator**: `registry.connect.redhat.com/redislabs/redis-enterprise-operator:7.2.4-8`
- **Services Rigger**: `registry.connect.redhat.com/redislabs/services-manager:7.2.4-8`
```

#### OLM bundle

> **Redis Enterprise operator bundle** : `v7.2.4-2`

Reviewer: ```suggestion
**Redis Enterprise operator bundle** : `v7.2.4-8`
```

### New limitations

No new known limitations were introduced in this version.

### Existing limitations

- **REAADB changes might fail with "gateway timeout" errors, mostly on OpenShift (RED-103048)** Retry the operation.

- **Misleading error appears when a service creation has failed due to service already existing "`provided IP is already allocated`" (RED-100669)** Delete the service manually.

- **Existing OpenShift route during REC creation might prevent REC from starting (RED-100668)** Delete the route manually.

- **Creating two databases with the same name directly on Redis Enterprise software will cause the service to be deleted and the database will not be available (RED-99997)** Avoid duplicating database names. Database creation via K8s has validation in place to prevent this.

- **Installing the operator bundle produces warning: `Warning: would violate PodSecurity "restricted: v1.24"` (RED-97381)** Ignore the warning. This issue is documented as benign on official Red Hat documentation.

> - **RERC resources must have a unique name (RED-96302)** The string "rec-name"/"rec-namespace" must be different from all other participating clusters in teh Active-Active database.

Reviewer: ```suggestion
- **RERC resources must have a unique name (RED-96302)** The string "rec-name"/"rec-namespace" must be different from all other participating clusters in the Active-Active database.
```

- **Deleting an OpenShift project with an REC deployed may hang (RED-47192)** When an REC cluster is deployed in a project (namespace) and has REDB resources, the REDB resources must be deleted first before the REC can be deleted. Therefore, until the REDB resources are deleted, the project deletion will hang. The fix is to delete the REDB resources first and the REC second. Then, you can delete the project.

- **Clusters must be named 'rec' in OLM-based deployments (RED-39825)** In OLM-deployed operators, the deployment of the cluster will fail if the name is not "rec". When the operator is deployed via the OLM, the security context constraints (scc) are bound to a specific service account name (namely, "rec"). The workaround is to name the cluster "rec".

- **Readiness probe incorrect on failures (RED-39300)** STS Readiness probe does not mark a node as "not ready" when running `rladmin status` on node failure.

- **Internal DNS and Kubernetes DNS may have conflicts (RED-37462)** DNS conflicts are possible between the cluster `mdns_server` and the K8s DNS. This only impacts DNS resolution from within cluster nodes for Kubernetes DNS names.

> - **K8s-based 5.4.10 clusters seem to negatively effect existing 5.4.6 clusters (RED-37233)** Upgrade clusters to latest version.

Reviewer: ```suggestion
- **K8s-based 5.4.10 clusters seem to negatively affect existing 5.4.6 clusters (RED-37233)** Upgrade clusters to latest version.
```

- **Node CPU usage is reported instead of pod CPU usage (RED-36884)** In Kubernetes, the reported node CPU usage is the usage of the Kubernetes worker node hosting the REC pod.

> - **An unreachable cluster has status running (RED-32805)** When a cluster is in unreachable state, the state remains `running` instead of triggering an error.

Reviewer: ```suggestion
- **An unreachable cluster has status running (RED-32805)** When a cluster is in an unreachable state, the state remains `running` instead of triggering an error.
```
